mopro-ffi/src/circom/mod.rs

```diff
@@ -150,18 +135,12 @@ pub fn verify_circom_proof(
     let file = File::open(&zkey_path).map_err(|e| MoproError::CircomError(e.to_string()))?;
     let mut reader = std::io::BufReader::new(file);
     if header_reader.r == BigUint::from(ark_bn254::Fr::MODULUS) {
-        let mut binfile = BinFile::<_, Bn254>::new(&mut reader)
-            .map_err(|e| MoproError::CircomError(e.to_string()))?;
-        let proving_key = binfile
-            .proving_key()
+        let proving_key = read_proving_key::<_, Bn254>(&mut reader)
```

Comment:
@vimwitch I use a `read_proving_key` in circom-compat
I think it is more readable function (than like `BinFile`)

Reply:
I agree, looks good